logger/log.go

```diff
@@ -0,0 +1,65 @@
+package logger
+
+import (
+	"fmt"
+	"runtime"
+	"time"
+
+	"github.com/fatih/color"
+)
+
+//Stores the colors once
+type logger struct {
+	c map[string]*color.Color
+}
+
+var instance *logger
+
+//Creates the Singleton Logger
+func InitLogger() {
+	//TODO Needs to accept a path to output the log to.
+	//TODO Maybe accept a logging level on what to display. That way, someone
+	//can run this on DEBUG mode and have all the DEBUG messages print out.
+	instance = &logger{}
+	instance.c = make(map[string]*color.Color)
+	instance.c["cyan"] = color.New(color.FgCyan)
+	instance.c["yellow"] = color.New(color.FgHiYellow)
+}
+
+//Returns an instance to the logger
+func GetLogger() *logger {
+	return instance
+}
+
+//A wrapper function for fmt.Printf. This function will
+//Look for the caller of the Logf function and automatically
+//print in the following format:
+//<timestamp><caller><Message>
+func (l *logger) Logf(format string, args ...interface{}) string {
+	//TODO Maybe accept logging levels/importance of the message
+	t := time.Now()
+	timeStamp := t.Format("2006-01-02 15:04:05.000")
+	caller := findCaller(3)
+	l.c["cyan"].PrintfFunc()("%s ", timeStamp)
+	l.c["yellow"].PrintfFunc()("%s: ", caller)
+	logString := fmt.Sprintf(format, args...)
+	fmt.Print(logString)
+	return logString
+}
+
+//1 = itself
+//2 = 1 above (this would be Logf)
+//3 = 2 above etc.
+func findCaller(callstack int) string {
```

Comment:
Well, this function is a private function and it will stay private (so no one else should/can use it). Additionally, it only accepts a (callstack int) because I wanted to be able to test it. 

I could remove it if you want (since I will be doing only the /api, so all my debug messages will come from only there). But when I was creating a website with multiple pages and handlers, It was really useful to just be able to "log("something"), and the output would be in the format of like..

`2017-03-18 10:37:28.740 github.com/ENEL500/controllers.DatabaseConnect: Connected to:10.1.21-MariaDB-1~xenial`
`2017-03-18 10:37:47.753 github.com/ENEL500/controllers.LoginController: Login Page
2017-03-18 10:37:47.753 github.com/ENEL500/controllers.LoginController: method:GET
2017-03-18 10:37:47.753 github.com/ENEL500/controllers.RenderTemplate: Rendering template: login.html
`
It helps you follow the flow of the website if you need to really debug it before the panic.